packages/mui-material/src/MobileStepper/MobileStepper.js

```diff
@@ -33,40 +35,65 @@ const MobileStepperRoot = styled(Paper, {
 
     return [styles.root, styles[`position${capitalize(ownerState.position)}`]];
   },
-})(({ theme, ownerState }) => ({
+})(({ theme }) => ({
   display: 'flex',
   flexDirection: 'row',
   justifyContent: 'space-between',
   alignItems: 'center',
   background: (theme.vars || theme).palette.background.default,
   padding: 8,
-  ...(ownerState.position === 'bottom' && {
-    position: 'fixed',
-    bottom: 0,
-    left: 0,
-    right: 0,
-    zIndex: (theme.vars || theme).zIndex.mobileStepper,
-  }),
-  ...(ownerState.position === 'top' && {
-    position: 'fixed',
-    top: 0,
-    left: 0,
-    right: 0,
-    zIndex: (theme.vars || theme).zIndex.mobileStepper,
-  }),
+  variants: [
+    {
+      props: { position: 'bottom' },
+      style: {
+        borderRadius: 0,
+        position: 'fixed',
+        bottom: 0,
+        left: 0,
+        right: 0,
+        zIndex: (theme.vars || theme).zIndex.mobileStepper,
+      },
+    },
+    {
+      props: { position: 'top' },
+      style: {
+        position: 'fixed',
+        top: 0,
+        left: 0,
+        right: 0,
+        zIndex: (theme.vars || theme).zIndex.mobileStepper,
+      },
```

Comment:
Some of the common styles can be moved to a common variant like,

```js
{
  props: ({position}) => position !== 'static',
  style: {
    position: 'fixed',
    left: 0,
    right: 0,
    zIndex: (theme.vars || theme).zIndex.mobileStepper,
  }
}
```
This'll leave position `bottom` with only `bottom: 0` and `top` with only `top: 0`